### DISCUSSION TITLE
Disable Cache

### DISCUSSION THREAD

**Discussion by folson-csi** [nginxinc/nginx-s3-gateway#123] [Ideas]
I have a project that is retrieving images from my S3 bucket where the images update about every 10 minutes. With the current implementation of this docker image, I am not able to find any way to disable the caching. Currently I am seeing the images only being update every hour or so. Is there a way to disable the cache completely? I've tried setting the PROXY_CACHE_VALID_OK to 1m. It would be useful if I could disable caching as a whole via an ENV variable.

**Comment by gimse** [comment]
I also need this function. The cache works bad with using byte range.

**Comment by 4141done** [comment]
Thank you for bringing this up and I apologize for the delay in my response.  I just tested and it appears that you can set `PROXY_CACHE_VALID_OK` to `0` and you will effectively turn off the cache although it's not well documented in the docs.

I tested by setting this and uncommenting `add_header X-Cache-Status $upstream_cache_status;` in my build of the s3 gateway image.  It'll show `X-Cache-Status: MISS` for all previously uncached requests.  For previously cached requests you'll have to wait for the TTL to expire. Please let me know if this fits your need or if I am misunderstanding.

I think also we can consider other solutions to this when looking at https://github.com/nginxinc/nginx-s3-gateway/issues/188 but I would really like to avoid adding more env var settings unnecessarily.  If setting that value to `0` works for you then we can improve the documentation.

**Reply by mikejavadivastspace** [reply]
That achieved the functionality I was after. thank you
